src/redux/history/reducers.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Reducer } from 'redux'
+import {
+  AddEntryAction,
+  HistoryAction,
+  HistoryActionType,
+  HistoryEntry,
+  RemoveEntryAction,
+  SetEntriesAction,
+  UpdateEntryAction
+} from './types'
+import { mergeHistoryEntries } from './methods'
+
+// The history reducer will be created without entries because of async entry retrieval.
+// Entries will be added after reducer initialization.
```

Comment:
One could ask: "Why is the reducer initialized with an empty array instead of the actual history entries like it is done for the config reducer?".
This comment responds that, this isn't possible because the history entry retrieval uses async functions that may not be used inside the intializer of the reducer.

The solution therefore is to add the history entries after the reducer is initialized.

Reply:
Well. I never asked myself that question :D I don't know if it's necessary. But if you think it is then at least add the question as well.

Reply:
Added the question as well.